File: android/ac-teams/src/main/kotlin/com/microsoft/adaptivecards/teams/TeamsCardHost.kt
package com.microsoft.adaptivecards.teams

import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import com.microsoft.adaptivecards.core.models.AdaptiveCard

@Composable
fun TeamsCardHost(
    card: AdaptiveCard,
    theme: TeamsTheme = TeamsTheme.LIGHT,
    tokenProvider: AuthTokenProvider? = null,
    deepLinkHandler: DeepLinkHandler = DeepLinkHandler(),
    content: @Composable () -> Unit
) {
    MaterialTheme(
        colorScheme = TeamsFluentTheme.getColorScheme(theme)
    ) {
        content()
    }
}

File: ios/Sources/ACTeams/TeamsCardHost.swift
import SwiftUI
import ACCore
import ACRendering

public struct TeamsCardHost<Content: View>: View {
    let card: AdaptiveCard
    let theme: TeamsTheme
    let tokenProvider: AuthTokenProvider?
    let deepLinkHandler: DeepLinkHandler
    let content: () -> Content
    
    @StateObject private var taskModulePresenter = TaskModulePresenter()
    @StateObject private var stageViewPresenter = StageViewPresenter()
    
    public init(
        card: AdaptiveCard,
        theme: TeamsTheme = .light,
        tokenProvider: AuthTokenProvider? = nil,
        deepLinkHandler: DeepLinkHandler = DeepLinkHandler(),
        @ViewBuilder content: @escaping () -> Content
    ) {
        self.card = card
        self.theme = theme
        self.tokenProvider = tokenProvider
        self.deepLinkHandler = deepLinkHandler
        self.content = content
    }
    
    public var body: some View {
        content()
            .preferredColorScheme(colorScheme)
            .sheet(isPresented: $taskModulePresenter.isPresented) {
                if let url = taskModulePresenter.url {
                    TaskModuleView(
                        url: url,
                        title: taskModulePresenter.title,
                        onDismiss: { taskModulePresenter.dismiss() }
                    )
                }
            }
            .fullScreenCover(isPresented: $stageViewPresenter.isPresented) {
                if let url = stageViewPresenter.url {
                    StageView(
                        url: url,
                        title: stageViewPresenter.title,
                        onDismiss: { stageViewPresenter.dismiss() }
                    )
                }
            }
    }
    
    private var colorScheme: ColorScheme? {
        switch theme {
        case .light:
            return .light
        case .dark, .highContrast:
            return .dark
        }
    }
}
